apps/protocol/src/app/hooks/useEstimatedOutputMulti.ts

```diff
@@ -37,52 +36,48 @@ interface Output {
  * This hook is designed for use with contracts that support mint & mintMulti, redeemExact
  */
 export const useEstimatedOutputMulti = (
+  route: Route,
+  scaledInput: ScaledInput,
+  lpPriceAdjustment?: LPPriceAdjustment,
   contract?: MintableContract,
-  inputValues?: BigDecimalInputValues,
-  lpPriceAdjustment?: { price: BigDecimal; isInput: boolean },
-  route?: Route,
 ): Output => {
   const [estimatedOutputRange, setEstimatedOutputRange] = useFetchState<[BigDecimal, BigDecimal]>()
-  const massetConfig = useSelectedMassetConfig()
-
-  const touched = Object.values(inputValues ?? {}).filter(v => v.touched)
 
   const priceImpact = useMemo<FetchState<PriceImpact>>(() => {
-    if (estimatedOutputRange.fetching || !estimatedOutputRange.value) return { fetching: true }
-
-    if (!touched.length) return {}
-
-    const totalInputLow = touched.reduce(prev => prev.add(massetConfig.lowInputValue), BigDecimal.ZERO)
-    const scaledInputHigh = touched.reduce((prev, v) => (v.amount ? prev.add(v.amount.scale()) : prev), BigDecimal.ZERO)
+    if (estimatedOutputRange.fetching) return { fetching: true }
 
-    if (!scaledInputHigh.exact.gt(0)) return { fetching: true }
+    if (scaledInput.scaledHighTotal.exact.eq(0) || !estimatedOutputRange.value) return {}
 
-    const value = getPriceImpact([totalInputLow, scaledInputHigh], estimatedOutputRange.value, lpPriceAdjustment, route === Route.Redeem)
+    const value = getPriceImpact(
+      [scaledInput.scaledLowTotal, scaledInput.scaledHighTotal],
+      estimatedOutputRange.value,
+      lpPriceAdjustment,
+      route === Route.Redeem,
+    )
 
     return { value }
-  }, [estimatedOutputRange.fetching, estimatedOutputRange.value, touched, route, lpPriceAdjustment, massetConfig.lowInputValue])
+  }, [estimatedOutputRange.fetching, estimatedOutputRange.value, scaledInput, lpPriceAdjustment, route])
 
   const [update] = useDebounce(
     () => {
-      if (!contract || !touched.length) return {}
+      if (!contract || Object.values(scaledInput.values).length === 0) return {}
```

Comment:
@JamesLefrere `scaledInput.values` can be `undefined`

Reply:
It shouldn't be, but you made me check and found that it wasn't initialised in the `reduce` 👍